daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala

```diff
@@ -528,7 +528,7 @@ private[daml] class EncodeV1(minor: LV.Minor) {
     private def encodeExprBuilder(
         expr0: Expr,
         builder: PLF.Expr.Builder = PLF.Expr.newBuilder(),
-    ): PLF.Expr.Builder = {
+    ): builder.type = {
```

Comment:
This ensures the function actually returns `builder` and does not return a new one.

Reply:
I like this!